lib/tcpip_stack_direct.ml

```diff
@@ -77,10 +77,10 @@ struct
   let ipv4 { ipv4; _ } = ipv4
 
   let listen_udpv4 t ~port callback =
-    Hashtbl.replace t.udpv4_listeners port callback
+    Hashtbl.replace t.udpv4_listeners (port mod 65536) callback
```

Comment:
Agreed, let's go for forbidding incorrect ports rather than the wrapping behaviour.